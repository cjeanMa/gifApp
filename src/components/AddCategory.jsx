import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ addCategory }) => {
  const [inputCat, setInputCat] = useState("");

  const onInputChange = (e) => {
    setInputCat(e.target.value);
  };

  const onSubmit = (e) => {
    e.preventDefault();
    setInputCat("");
    if (inputCat.trim().length <= 1) return;
    addCategory(inputCat);
  };

  return (
    <form onSubmit={onSubmit} name="formCategory">
      <input
        type="text"
        placeholder="Buscar"
        value={inputCat}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.protoTypes = {
  addCategory: PropTypes.func.isRequired,
};
