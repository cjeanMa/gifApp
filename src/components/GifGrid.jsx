import React, { useEffect, useState } from "react";
import getGifs from "../clients/getGifs";

const GifGrid = ({ category }) => {
  const [gifs, setGifs] = useState([]);

  const gettingGifs = async (category) => {
    const gifsApi = await getGifs(category);
    setGifs(gifsApi);
  };

  useEffect(() => {
    gettingGifs(category);
  }, []);

  return (
    <>
      <h3>{category}</h3>
      <ol>
        {gifs.map((el) => (
          <li key={el.id}>{el.title}</li>
        ))}
      </ol>
      ;
    </>
  );
};

export default GifGrid;
