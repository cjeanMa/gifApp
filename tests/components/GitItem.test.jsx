import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Test component GifItem", () => {
  const title = "Canon";
  const url = "https://facebook.com/profile.jpg";
  test("test snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("it should show the image with url and alt indicated", () => {
    render(<GifItem title={title} url={url} />);
    // expect(screen.getByRole("img").src).toBe(url);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("it should show the title", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
