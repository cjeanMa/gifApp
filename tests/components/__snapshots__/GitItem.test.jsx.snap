// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component GifItem test snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Canon"
      src="https://facebook.com/profile.jpg"
    />
    <p>
      Canon
    </p>
  </div>
</div>
`;
